awx/main/models/unified_jobs.py

```diff
@@ -842,15 +843,18 @@ def save(self, *args, **kwargs):
             if 'finished' not in update_fields:
                 update_fields.append('finished')
 
+        dq = decimal.Decimal('1.000')
+        if self.elapsed is None:
+            self.elapsed = decimal.Decimal(0.0).quantize(dq)
+            if 'elapsed' not in update_fields:
+                update_fields.append('elapsed')
```

Comment:
I don't understand why you would want to add elapsed to update fields here. You don't get into this unless `self.elapsed` is None, but why would that happen? If the job isn't finished, I don't want to update the field, even if it's setting it to 0.

Maybe I could see why you'd want to initialize it to a `decimal.Decimal` object, but why do so only for the case where it's unset?

Reply:
yeah I think update_fields is not needed in the `self.elapsed` `None` case. The `elapsed` DecimalField is non-null, so I think the column will be updated in this case regardless? Not sure what the consequence of having `update_fields` here, but it works without it so I'll remove it.

> Maybe I could see why you'd want to initialize it to a decimal.Decimal object, but why do so only for the case where it's unset?

I cast `self.elapsed` to Decimal in both cases -- where it's initialized to 0.0, and below where it's calculated based on `started` and `finished`

Reply:
but really converting elapsed to Decimal isn't going to prevent the awx_collections tests from breaking as I had hoped. If people would rather return a float here I'm okay with that too